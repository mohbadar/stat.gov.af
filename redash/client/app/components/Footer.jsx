import React from 'react';
import PropTypes from 'prop-types';

import { react2angular } from 'react2angular';

import frontendVersion from '../version.json';

export function Footer({ clientConfig, currentUser }) {
  const backendVersion = clientConfig.version;
  const newVersionAvailable = clientConfig.newVersionAvailable && currentUser.isAdmin;
  const separator = ' \u2022 ';

  let newVersionString = '';
  if (newVersionAvailable) {
    newVersionString = (
      <small>
        <a href="https://version.redash.io/">(New Redash version available)</a>
      </small>
    );
  }

  return (
    <div id="footer">
      <a href="https://redash.io">Redash</a> {backendVersion} ({frontendVersion.substring(0, 8)})
      {newVersionString}
      {separator}
      <a href="https://redash.io/help/">Documentation</a>
      {separator}
      <a href="https://github.com/getredash/redash">Contribute</a>
    </div>
  );
}

Footer.propTypes = {
  clientConfig: PropTypes.shape({
    version: PropTypes.string,
    newVersionAvailable: PropTypes.bool,
  }).isRequired,
  currentUser: PropTypes.shape({
    isAdmin: PropTypes.bool,
  }).isRequired,
};

export default function init(ngModule) {
  ngModule.component('footer', react2angular(Footer, [], ['clientConfig', 'currentUser']));
}

init.init = true;
