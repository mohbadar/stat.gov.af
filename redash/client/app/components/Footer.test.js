import React from 'react';
import renderer from 'react-test-renderer';
import { Footer } from './Footer';

test('Footer renders', () => {
  const clientConfig = {
    version: '5.0.1',
    newVersionAvailable: true,
  };
  const currentUser = {
    isAdmin: true,
  };
  const component = renderer.create(<Footer clientConfig={clientConfig} currentUser={currentUser} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
