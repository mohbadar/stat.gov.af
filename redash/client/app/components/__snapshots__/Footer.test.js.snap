// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
<div
  id="footer"
>
  <a
    href="https://redash.io"
  >
    Redash
  </a>
   
  5.0.1
   (
  dev
  )
  <small>
    <a
      href="https://version.redash.io/"
    >
      (New Redash version available)
    </a>
  </small>
   • 
  <a
    href="https://redash.io/help/"
  >
    Documentation
  </a>
   • 
  <a
    href="https://github.com/getredash/redash"
  >
    Contribute
  </a>
</div>
`;
